app/src/main/java/com/nutomic/syncthingandroid/model/RunConditionCheckResult.java

```diff
@@ -0,0 +1,68 @@
+package com.nutomic.syncthingandroid.model;
+
+import com.nutomic.syncthingandroid.R;
+
+import java.util.Collections;
+import java.util.List;
+
+public class RunConditionCheckResult {
+
+    public enum BlockerReason {
+        ON_BATTERY(R.string.syncthing_disabled_reason_on_battery),
+        ON_CHARGER(R.string.syncthing_disabled_reason_on_charger),
+        POWERSAVING_ENABLED(R.string.syncthing_disabled_reason_powersaving),
+        GLOBAL_SYNC_DISABLED(R.string.syncthing_disabled_reason_android_sync_disabled),
+        WIFI_SSID_NOT_WHITELISTED(R.string.syncthing_disabled_reason_wifi_ssid_not_whitelisted),
+        WIFI_WIFI_IS_METERED(R.string.syncthing_disabled_reason_wifi_is_metered),
+        NO_NETWORK_OR_FLIGHTMODE(R.string.syncthing_disabled_reason_no_network_or_flightmode),
+        NO_MOBILE_CONNECTION(R.string.syncthing_disabled_reason_no_mobile_connection),
+        NO_WIFI_CONNECTION(R.string.syncthing_disabled_reason_no_wifi_connection),
+        NO_ALLOWED_NETWORK(R.string.syncthing_disabled_reason_no_allowed_method);
+
+        private final int resId;
+
+        BlockerReason(int resId) {
+            this.resId = resId;
+        }
+
+        public int getResId() {
+            return resId;
+        }
+    }
+
+    private final boolean mShouldRun;
+    private final List<BlockerReason> mBlockReasons;
+
+    public RunConditionCheckResult() {
+        mBlockReasons = Collections.emptyList();
+        mShouldRun = true;
+    }
+
+    public RunConditionCheckResult(List<BlockerReason> blockReasons) {
+        mBlockReasons = Collections.unmodifiableList(blockReasons);
+        mShouldRun = blockReasons.isEmpty();
+    }
+
+
+    public List<BlockerReason> getBlockReasons() {
+        return mBlockReasons;
+    }
+
+    public boolean isShouldRun() {
+        return mShouldRun;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        RunConditionCheckResult that = (RunConditionCheckResult) o;
+        return mShouldRun == that.mShouldRun && mBlockReasons.size() == that.mBlockReasons.size()
+                && mBlockReasons.containsAll(that.mBlockReasons);
+    }
```

Comment:
IntelliJ (and potentially Android studio) can generate both hashcode and equals whcih has a better implementation.
It seems strange that equality takes reasons into account, and hash code does not.

Reply:
You are looking at the old version. This is already outdated by my latest push. This version has been replaced by the InteliJ version.

Reply:
```
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        RunConditionCheckResult that = (RunConditionCheckResult) o;

        if (mShouldRun != that.mShouldRun) return false;
        return mBlockReasons.equals(that.mBlockReasons);
    }

    @Override
    public int hashCode() {
        int result = (mShouldRun ? 1 : 0);
        result = 31 * result + mBlockReasons.hashCode();
        return result;
    }
```

is what I see. Usually IntelliJ generates `Object.equals(x, y) && ...` etc, and `Object.hash(o) + Object.hash(y)`

Reply:
I needed to the revert to the default generated version (you can change that in the generate dialog), since our target API level does not have `Object.hash(o)` available. This is available from API 19, we are still targeting API 16.